src/python/WMCore/Storage/Backends/XRDCPImpl.py

```diff
@@ -59,17 +101,44 @@ def _checkXRDUtilsExist(self):
 
         return foundXrdcp & foundXrdfs
 
-    def createStageOutCommand(self, sourcePFN, targetPFN, options=None, checksums=None):
+    def getAuthEnv(self, authMethod=None, forceMethod=False):
         """
-        _createStageOutCommand_
+        Get environment variables for stageout command based on the selected authentication method.
+        :authMethod: str, the authentication method to be used ("X509", "TOKEN", or None)
+        :forceMethod: bool, cleans non-chosen auth methods from environment.
+        :return: str
+        """
+        authEnv = ""
+        if authMethod is None:
+            return authEnv
+        elif authMethod.upper() == 'X509':
+            authEnv = self.setAuthX509
+            if forceMethod:
+                authEnv += self.unsetToken
+        elif authMethod.upper() == 'TOKEN':
+            authEnv = self.setAuthToken
+            if forceMethod:
+                authEnv += self.unsetX509
+        else:
+            logging.info("Warning! Running without either a X509 certificate or a token specified!")
 
-        Build the actual xrdcp stageout command
+        return authEnv
 
+    def getFormattedCopyCommand(self, sourcePFN, targetPFN, options=None, checksums=None, authMethod=None, forceMethod=False, debug=False):
+        """
+        Construct xrdcp/xrdfs stageout command.
         If adler32 checksum is provided, use it for the transfer
         xrdcp options used:
           --force : re-creates a file if it's already present
           --nopbar : does not display the progress bar
 
+        :sourcePFN: str, the source PFN.
+        :targetPFN: str, the target PFN.
+        :options: str, additional options for gfal-copy.
+        :checksums: dict, checksum values.
+        :authMethod: str, preferred authentication method.
+        :forceMethod: bool, whether to force the preferred authentication method.
+        :debug: bool, enable/disable debugging mode after command.
```

Comment:
I would suggest to keep things consistent with the gfal implementation. Hence either adopt `dryRun` variable here, or `debug` in the gfal one please.